### PR TITLE
EVG-13094 Jest tests for the build baron tab

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

**Review by khelif96 — CHANGES_REQUESTED** [review]
This is a great start! left some suggestions. The MutationObserver pollyfill does not seem necessary i was able to remove it with out any errors. There are some warnings being thrown but i believe those could be resolved by upgrading testing libraries.

File: src/hooks/useBuildBaronVariables.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const useBuildBaronVariables = ({
   const buildBaronConfigured = buildBaron?.buildBaronConfigured;
 
   // logic for displaying the build baron tab
-  const buildBaronIsProductionReady = false;
+  const buildBaronIsProductionReady = true;
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply src/hooks/useBuildBaronVariables.ts:42]: It's been ready and waiting on tests

File: src/components/Buildbaron/BuildBaron.test.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  {
+    request: {
+      query: GET_USER,
+    },
+    result: {
+      data: {
+        userId: "mohamed.khelif",
+        displayName: "Mohamed Khelif",
+      },
+    },
```
Comment (khelif96) [inline src/components/Buildbaron/BuildBaron.test.tsx:174]: I think this request was mocked twice (on line 137 also), Is this intentional?

File: src/components/Buildbaron/BuildBaron.test.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  expect(queryByDataCy("EVG-12346-badge")).toHaveTextContent("Closed");
+  expect(queryByDataCy("EVG-12346-metadata")).toHaveTextContent(
+    "Created: Sep 18, 2020 Updated: Sep 18, 2020 Assignee: Some Name"
+  );
+
+  expect(queryByDataCy("EVG-12347-badge")).toHaveTextContent("Open");
+  expect(queryByDataCy("EVG-12347-metadata")).toHaveTextContent(
+    "Created: Sep 18, 2020 Updated: Sep 18, 2020 Assignee: Backlog - Evergreen Team"
+  );
+});
```
Comment (khelif96) [inline src/components/Buildbaron/BuildBaron.test.tsx:263]: Would it be possible to capture this functionality in a test. To ensure the polling for a recently created ticket works. https://github.com/evergreen-ci/spruce/blob/5133e34d05a2cd50429103521fa3608cb41f4469/src/components/Buildbaron/BBCreatedTickets.tsx#L45-L50

File: src/components/Buildbaron/BuildBaron.test.tsx (hunk 2 above)
Comment (malikchaya2) [reply src/components/Buildbaron/BuildBaron.test.tsx:174]: removed it, thanks!

File: src/components/Buildbaron/BuildBaron.test.tsx (hunk 3 above)
Comment (malikchaya2) [reply src/components/Buildbaron/BuildBaron.test.tsx:263]: discussed over slack and decided not to do

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**